spec.md

```diff
@@ -157,6 +157,10 @@ The `Content-Type` header SHOULD match what the client [pushed as the manifest's
 If the manifest has a `mediaType` field, clients SHOULD reject unless the `mediaType` field's value matches the type specified by the `Content-Type` header.
 For more information on the use of `Accept` headers and content negotiation, please see [Content Negotiation](./content-negotiation.md).
 
+The client SHOULD include a `Docker-Manifest-Tag` header indicating which manifest tag is being pulled.
```

Comment:
This shouldn't be `Docker` for a new header. And the manifest tag is probably not useful enough. From the linked issue, a full reference would be more appropriate, and there may be multiple headers for the reference to a multi-platform manifest plus the single platform manifest.

Reply:
1. As suggested - will change to use `OCI` in the header name instead of `Docker`: `OCI-Manifest-Tag`
2. The proposed header includes only the tag and not the full reference (`<registry>/<image>:<tag>`) because the `<image>` portion is already in the endpoints URL, and most likely also the registry as the `Host` (but not guaranteed). But I agree it is OK to require the full reference - as it serves the end goal. And it so, the header can be as suggested in #328 : `OCI-Ref`, or `OCI-Reference` (because I think it is OK to be more explicit and verbose in header names).
3. The original thought was to add also a header with the digests of the manifests in the chain - the image index(es) and the actual image manifest, but decided to keep it simple...

---

For example - based on the notes above, pulling an image with manifest list by ref `myreg.org/foo/bar:1.2` can include the following requests with headers:
1) Resolve the digest of the first manifest by tag (i.e. the image index):
```
HEAD /v2/foo/bar/manifests/1.2
OCI-Reference: myreg.org/foo/bar:1.2
```
2) Get the image index manifest by the digest that was returned in the response header to the request above:
```
GET /v2/foo/bar/manifests/sha256:abc123
OCI-Reference: myreg.org/foo/bar:1.2
OCI-Manifest-Digests: sha256:abc123
```
3) Get the actual manifest by the digest of the platform specific manifest (extracted from the image index json):
```
GET /v2/foo/bar/manifests/sha256:bcd234
OCI-Reference: myreg.org/foo/bar:1.2
OCI-Manifest-Digests: sha256:abc123 sha256:bcd234
```
4) Get the blobs as listed in the manifest:
```
GET /v2/foo/bar/blobs/sha256:def789
OCI-Reference: myreg.org/foo/bar:1.2
OCI-Manifest-Digests: sha256:abc123 sha256:bcd234
```
As seen above, the reference header is added to all requests, the digests are added along the way. When a blob or manifest is downloaded, the registry can know the context of the tagged image reference and also the digests of all the manifests in the chain.